reframe/core/buildsystems.py

```diff
@@ -782,6 +782,72 @@ def generated_modules(self):
         return self._eb_modules
 
 
+class Spack(BuildSystem):
+    '''A build system for building test code using `Spack
+    <https://spack.io/>`__.
+
+    ReFrame will use Spack to build and install the code given an environment
+    and a set of specs.
+
+    .. versionadded:: 3.6.0
+
+    '''
+
+    #: The list of specs to build and install within the given environment.
+    #: This field is required.
+    #:
+    #: :type: :class:`List[str]`
+    #: :default: ``[]``
+    specs = fields.TypedField(typ.List[str])
+
+    #: Spack environment.
+    #: ReFrame activates this environment to build and install the specified
+    #: packages.
+    #: This field is required.
+    #:
+    #: :type: :class:`str` or :class:`None`
+    #: :default: :class:`None`
+    environment = fields.TypedField(str, type(None))
+
+    def __init__(self):
+        super().__init__()
+        self.specs = []
+        self.environment = None
+        self._prefix_save = None
+
+    def emit_build_commands(self, environ):
+        self._prefix_save = os.getcwd()
+        if not self.environment:
+            raise BuildSystemError(f"'environment' must not be empty")
```

Comment:
You can avoid this check by setting the type of the `environment` field to `typ.Str[r'.+']`.